File: mobile-client/android/UseCase/Message/Thread/ListMessageItemsUseCase.kt
package app.mobiling.client.usecase.message.thread

import app.mobiling.client.contract.message.thread.MessageItemPayload
import app.mobiling.client.data.message.thread.MessageThreadGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class ListMessageItemsUseCase(
    private val gateway: MessageThreadGateway,
) {
    suspend operator fun invoke(threadId: String): List<MessageItemPayload> = gateway.listItems(threadId)
}

File: client/ios/UseCase/Message/Thread/ListMessageItemsUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct ListMessageItemsUseCase: Sendable {
    private let gateway: MessageThreadGateway

    public init(gateway: MessageThreadGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(threadId: String) async throws -> [MessageItemPayload] {
        try await gateway.listItems(threadId: threadId)
    }
}
